exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Table.kt

```diff
@@ -939,11 +939,49 @@ open class Table(name: String = "") : ColumnSet(), DdlAware {
      * when using the PostgreSQL dialect is allowed, but this value will be ignored by the database.
      * @throws IllegalStateException If no column type mapping is found.
      */
-    inline fun <reified E : Any> array(name: String, maximumCardinality: Int? = null): Column<List<E>> {
+    inline fun <reified E : Any> array(name: String, maximumCardinality: Int? = null): Column<List<E>> =
+        array<E, List<E>>(name, maximumCardinality?.let { listOf(it) }, dimensions = 1)
+
+    /**
+     * Creates a multi-dimensional array column, with the specified [name], for storing elements of a nested `List`.
+     * The number of dimensions is specified by the [dimensions] parameter.
+     *
+     * **Note:** This column type is only supported by PostgreSQL dialect.
+     *
+     * @param name Name of the column.
+     * @param maximumCardinality The maximum cardinality (number of allowed elements) for each dimension in the array.
+     * @param dimensions The number of dimensions of the array.
+     *
+     * **Note:** Providing an array size limit when using the PostgreSQL dialect is allowed, but this value will be ignored by the database.
+     *
+     * @return A column instance that represents a multi-dimensional list of elements of type [T].
+     * @throws IllegalArgumentException If [dimensions] is less than or equal to 1.
```

Comment:
Maybe I missed it, but I can't find anywhere that would throw under these specified conditions.